### ISSUE TITLE
C420 false negative at the start of input for builtin static values

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#15830]
### Description

[`unnecessary-dict-comprehension-for-iterable` (C420)](https://docs.astral.sh/ruff/rules/unnecessary-dict-comprehension-for-iterable/) has a false negative in Ruff 0.9.3 when the static value is a builtin and the comprehension starts at the very beginning of the input.
```console
$ printf '{x: NotImplemented for x in "XY"}\n' | ruff --isolated check --preview --select C420 - --output-format=concise
All checks passed!
```

If the value is not a builtin, the rule works.
```console
$ printf '{x: 1 for x in "XY"}\n' | ruff --isolated check --preview --select C420 - --output-format=concise --fix -q
dict.fromkeys("XY", 1)

$ printf '{x: None for x in "XY"}\n' | ruff --isolated check --preview --select C420 - --output-format=concise --fix -q
dict.fromkeys("XY")
```

If even a single character appears before the comprehension, the rule works.
```console
$ printf '\n{x: NotImplemented for x in "XY"}\n' | ruff --isolated check --preview --select C420 - --output-format=concise --fix -q

dict.fromkeys("XY", NotImplemented)

$ printf '({x: NotImplemented for x in "XY"})\n' | ruff --isolated check --preview --select C420 - --output-format=concise --fix -q
(dict.fromkeys("XY", NotImplemented))
```

**Comment by dylwil3** [comment]
What a weird one... but I think I figured it out 😄
